### PR TITLE
Add close event

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

**Comment by LuKks** [comment]
@mafintosh Ready!

File: index.js [hunk 1]
```diff
@@ -222,9 +222,12 @@ class DHT extends EventEmitter {
   }
 
   destroy () {
+    if (this.destroyed) return
     this.destroyed = true
     clearInterval(this._tickInterval)
-    return this.io.destroy()
+    const p = this.io.destroy()
+    p.finally(() => this.emit('close'))
+    return p
```
Comment (mafintosh) [inline index.js:230]: just make destroy async and await destroy, nicer

File: index.js [hunk 2]
```diff
@@ -221,10 +221,12 @@ class DHT extends EventEmitter {
     this._backgroundQuery(node ? node.id : this.table.id).on('error', noop)
   }
 
-  destroy () {
+  async destroy () {
+    if (this.destroyed) return
```
Comment (mafintosh) [inline index.js:225]: let the other destroys through still actually, the io.destroy streamlines them

but save a flag if this is the first one and the only emit close at the btm if its the first one

**Comment by mafintosh** [comment]
6.5.0

**Pull request merged**